File: php/src/ItemUpdater/ConjuredItemUpdater.php
<?php

declare(strict_types=1);

namespace GildedRose\ItemUpdater;

use GildedRose\Item;
use GildedRose\ItemUpdater\Abstract\AbstractItemUpdater;
use GildedRose\ItemUpdater\Interface\ItemUpdaterInterface;

class ConjuredItemUpdater extends AbstractItemUpdater implements ItemUpdaterInterface
{
    /**
     * @param Item $item
     *
     * @return void
     */
    public function updateQuality(Item $item): void
    {
        // "Conjured" items degrade in Quality twice as fast as normal items
        $this->decreaseQuality($item, $this->getQualityDecrease($item) * 2);
    }
}

File: Java/src/main/java/com/gildedrose/updaters/ConjuredItemUpdater.java
package com.gildedrose.updaters;

import com.gildedrose.Item;

public class ConjuredItemUpdater extends ItemUpdater {
    public ConjuredItemUpdater(Item item) {
        super(item);
    }

    @Override
    public void update() {
        decreaseQuality(2);
        decreaseSellIn();
        if (item.sellIn < 0) {
            decreaseQuality(2);
        }
    }
}
